### PR TITLE
98 items everywhere

### PR DESCRIPTION
**Pull request by FBroy**
# Description

We returned undefined in cases we didn't know what to annotate, now, after discussing this with Martin, we return item()*.
There is also a TODO in typeswitchOperator that has been fixed.

## List of changes

- Fix TODO in typeswitchOperator
- Fix documentation throughout the annotation files
- Remove unused parameters from functions
- Return item()* where we don't know the exact type but we don't annotate the AST in this case

## Deletes Source Branch?

Yes:    Work is done, the branch can be safely deleted.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/typeInference/annotateTypeSwitchOperator.ts [hunk 1]
```diff
@@ -48,23 +49,22 @@ function checkComparison(
 	argumentType: SequenceType,
 	returnType: SequenceType
 ): SequenceType | undefined {
```
Comment (HectorPeeters) [inline src/typeInference/annotateTypeSwitchOperator.ts:86]: This undefined can go right?

**Review by HectorPeeters — APPROVED** [review]
Looks good!

File: src/typeInference/annotateTypeSwitchOperator.ts (hunk 1 above)
Comment (FBroy) [reply src/typeInference/annotateTypeSwitchOperator.ts:86]: Nope, I use it in the actual annotation function.

**Review by MTimmerije — APPROVED** [review]
Nice PR

**Review by MTimmerije — APPROVED** [review]
Looks pretty good, finally the test set is passing again.

File: src/expressions/dataTypes/casting/tryCastToType.ts [hunk 2]
```diff
@@ -26,18 +26,20 @@ import castToGYear from './castToGYear';
...
 import castToHexBinary from './castToHexBinary';
 import castToInteger from './castToInteger';
+import castToNumeric from './castToNumeric';
 import castToString from './castToString';
 import castToTime from './castToTime';
 import castToUntypedAtomic from './castToUntypedAtomic';
 import castToYearMonthDuration from './castToYearMonthDuration';
 
 const TREAT_AS_PRIMITIVE = [
+	ValueType.XSNUMERIC,
```
Comment (HectorPeeters) [inline src/expressions/dataTypes/casting/tryCastToType.ts:36]: Are we sure this is a primitive or was this added as a temporary fix?

Comment (FBroy) [reply src/expressions/dataTypes/casting/tryCastToType.ts:36]: Numeric is a union type, so this is a temporary fix. I'll make it an issue

Comment (HectorPeeters) [reply src/expressions/dataTypes/casting/tryCastToType.ts:36]: Perfect!

**Review by HectorPeeters — APPROVED** [review]

**Pull request merged**